src/components/Toast/Toast.stories.tsx

```diff
@@ -25,37 +26,55 @@ export const Basic: Story = {
   render: function Render() {
     const { showToast } = useToastStore();
 
-    useEffect(() => {
-      showToast({ message: '테스트', type: TOAST_DURATION_TIME.SHOW });
-
-      const interval = setInterval(
-        () => showToast({ message: '테스트', type: TOAST_DURATION_TIME.SHOW }),
-        TOAST_DURATION_TIME.SHOW + 1000,
-      );
-
-      return () => clearInterval(interval);
-    }, [showToast]);
-
-    return <Toast />;
+    return (
+      <>
+        <Button
+          state="enabled"
+          size="M"
+          onClick={() => showToast({ message: '일반 토스트 노출' })}
+        >
+          일반 토스트 노출
+        </Button>
+        <Toast />
+      </>
+    );
+  },
+  parameters: {
+    docs: {
+      description: {
+        story: '사용자 액션이 없는 토스트 (노출시간: 3.5초)',
+      },
+    },
   },
 };
 
 export const WithAction: Story = {
   render: function Render() {
     const { showToast } = useToastStore();
 
-    useEffect(() => {
-      showToast({ message: '테스트', type: TOAST_DURATION_TIME.ACTION });
-
-      const interval = setInterval(
-        () =>
-          showToast({ message: '테스트', type: TOAST_DURATION_TIME.ACTION }),
-        TOAST_DURATION_TIME.ACTION + 1000,
-      );
-
-      return () => clearInterval(interval);
-    }, [showToast]);
-
-    return <Toast />;
+    return (
+      <>
+        <Button
+          state="enabled"
+          size="M"
+          onClick={() =>
+            showToast({
+              message: '액션이 있는 토스트 노출',
+              type: TOAST_DURATION_TIME.ACTION,
+            })
+          }
+        >
+          액션이 있는 토스트 노출
+        </Button>
+        <Toast />
+      </>
+    );
```

Comment:
```suggestion
   const { isOpen, onOpen } = useToast();
    return (
      <>
        <Button
          state="enabled"
          size="M"
          onClick={onOpen}
        >
          액션이 있는 토스트 노출
        </Button>
        {isOpen && <Toast duration={400}>액션이 있는 토스트 노출</Toast>}
      </>
    );
```
Button 컴포넌트가 Toast 컴포넌트의 내용들을 대신 관리하기 보다 해당 컴포넌트에서 관리해도 좋을 것 같아요. 
컴포넌트의 내용을 상태에 주입하고 꺼내는 일련의 과정들이 줄어들 것 같네요!

Reply:
@dmswl98 
요 부분 제가 정확히 이해했는지 모르겠는데, 현재 `onClick` props에 할당된 함수를 `useToast` 훅 내부로 분리하는 것을 말씀 주신게 맞을까요?

Reply:
개인적인 의견으로는 Button 컴포넌트는 Toast 컴포넌트의 노출의 여부만 관여하면 되지 않을까 싶었어요!
위 코드에서의 Toast 컴포넌트는 항상 스토어에 종속되어 있고, 외부에서 주입된 내용을 받아서 이루어지는 점에서 컴포넌트를 이루는 요소들이 분산되어 있다는 느낌이 들었습니다!

Reply:
@dmswl98 
Button 컴포넌트가 Toast 컴포넌트의 노출 여부만을 결정한다면, 그 내용은 혹시 어떻게 전달하는 형태가 될까요?

Reply:
```tsx
<Toast isOpen={isOpen} duration={400}>액션이 있는 토스트 노출</Toast>
`
``
내용은 이렇게 작성하면 되지 않을까 하는 개인적인 의견이었습니다!

Reply:
@dmswl98 
아아!!! 근데 실제로 컴포넌트 속에서 버튼 클릭을 통해 토스트를 노출시키는 경우의 코드는 다음과 같습니다.

```javascript
const Home = () => {

  return (
     <Button onClick={() => showToast({ message: '액션이 있는 토스트 노출' })}>
       액션이 있는 토스트 노출
     </Button>
  )
}
```
컴포넌트에서는 Toast 컴포넌트를 반환하지 않습니다. 스토리북에서는 스토리북의 특수성 때문에 Toast 컴포넌트를 반환하지 않는 경우 토스트가 노출되지 않는 이슈가 있어서 불필요하게 함께 반환하는 형태를 취했습니다.
그래서 현재의 형태를 유지해도 괜찮지 않을까 생각합니다.

Reply:
@dongkyun-dev 아하 이해했습니당 감사합니당 👏🏻